### ISSUE TITLE
Enhance logging when configured realms are not available in current license

### ISSUE THREAD

**Issue by jkakavas** [elastic/elasticsearch#45728]
We generally handle this case well and print relevant logs in `AuthenticationService#consumeUser` : 

```
logger.warn("Authentication failed using realms [{}]." +
                            " Realms [{}] were skipped because they are not permitted on the current license",
```

There are however cases when the `AuthenticationToken` extraction fails and we end up with a null token for the configured realms, in which cases we don't enter in `consumeUser` and never log this information. 

An example of this is when only the PKI realm is configured on a license that doesn't allow it ( i.e. Basic ) where we won't print anything in the logs, making it harder for users to troubleshoot the authentication error they get. 

I believe that the case with PKI is unique, as SAML and OIDC have relevant logging on the REST layer, but we should verify that adequate information is logged in all cases.

**Comment by sebgl** [comment]
(discuss topic where this came up recently: https://discuss.elastic.co/t/using-pki-based-auth-for-a-cluster-created-with-elastic-cloud-on-k8s/254245 - it took me a while to figure out the license problem)

**Comment by ywangd** [comment]
Wasn't aware of this issue when I raised #61090. They are in fact the same issue and are now resolved by #61402
